core/src/main/java/nl/nn/adapterframework/ldap/LdapClient.java

```diff
@@ -329,6 +330,10 @@ private String arrayToString(String[] values, String separator) {
 //    }
   
     public Set<String> searchRecursivelyViaAttributes(String uid, String baseDn, String attribute) throws NamingException {
+    	return searchRecursivelyViaAttributes(uid, baseDn, attribute, null);
+    }
+    
+    public Set<String> searchRecursivelyViaAttributes(String uid, String baseDn, String attribute, String stopCondition) throws NamingException {
```

Comment:
Maak je ergens ook gebruik van de mogelijkheid om een stopConditie mee te geven bij aanroep? Die zie ik namelijk nergens.

Reply:
Ja, vanuit de LdapFindMemberPipe geef ik de stopConditie mee indien recursief gezocht wordt (zie LdapFindMemberPipe.java, regel 70). De stopConditie is dan de dnFind waarvan bepaald moet worden of die lid is van de dnSearchIn waarbinnen gezocht wordt.

Reply:
De methode searchRecursivelyViaAttributes is bedoeld/ontwikkeld om op een snelle manier alle lidmaatschappen van een gebruiker op te kunnen halen. Het attribuut waar je dan op zoekt is 'memberOf'. Je haalt dan alle groepen op waar die gebruiker lid van is, en vervolgens, recursief, de groepen waar de gevonden groepen lid van zijn. Omdat het aantal groepen t.o.v. het aantal users beperkt is, en de lidmaatschapen van groepen niet snel wijzigen, kan die laatste slag gecached worden.
Wat hier nu gebeurt, is alle leden van groepen ophalen, en als die leden groepen zijn, daar weer de leden van ophalen.  Dat doe je om te bepalen of de gebruiker waarom het ging er tussen zit. Dit is geen effectieve manier van zoeken als de groepen veel leden hebben. Voorbeeld: Bepaal of gebruiker 'X' lid is van de groep 'Customers'. Je gaat dan hier door alle zeg 1 miljoen customer heen lopen, (en die records bewaren) om bij elke iteratie te kijken of de gebruiker die je zoek er al bij zit. 
Deze inefficientie zit niet in jouw nieuwe code, die zat al in de oude. De aanpassing aan LdapClient die je hier voorstelt helpt maar een heel klein beetje om het probleem op te lossen: je kan dan stoppen na typisch een half miljoen iteraties.
Ik zou willen voorstellen dat bij de LdapFindMemberPipe een waarschuwing komt die bovenstaand probleem duidelijk maakt, en dat je een nieuwe LdapIsMemberOfPipe maakt, die werkt via het 'memberOf' attribuut.

Reply:
Ik weet niet of een extra pipe daarvoor de meest gewenste oplossing is, omdat we dan 2 pipes hebben die eigenlijk hetzelfde doen. Of zou het toevoegen van een LdapIsMemberOfPipe een reden kunnen zijn om de LdapFindMemberPipe deprecated maken, omdat ze dezelfde functionaliteit bieden, alleen dan vanuit verschillende richtingen? Die zou dan wel een sterke overlap hebben met de LdapFindGroupMembershipsPipe, omdat de LdapFindGroupMembershipsPipe+pipe om te controleren of een bepaalde groep gevonden is eigenlijk dezelfde functionaliteit biedt, maar dan in 1 pipe gecombineerd.

Een alternatief is dat op de LdapFindMemberPipe een richting aangegeven kan worden of het attribuut dat afgelopen wordt ingesteld kan worden (waarmee de logica omgedraaid zou kunnen worden door searchIn en Find om te wisselen en het attribuut op memberOf te zetten).